LanguageFeatures/Inline-classes/syntax_A02_t05.dart

```diff
@@ -2,52 +2,41 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion A rule for <viewDeclaration> is added to the grammar, along with
-/// some rules for elements used in view declarations:
+/// @assertion A rule for <inlineClassDeclaration> is added to the grammar,
+/// along with some rules for elements used in inline class declarations:
 ///
-/// <viewDeclaration> ::=
-///   'view' 'class' <typeIdentifier> <typeParameters>?
-///       <viewPrimaryConstructor>?
-///       <interfaces>?
+/// <inlineClassDeclaration> ::=
+///   'inline' 'class' <typeIdentifier> <typeParameters>? <interfaces>?
 ///   '{'
-///     (<metadata> <viewMemberDeclaration>)*
+///     (<metadata> <inlineMemberDeclaration>)*
 ///   '}'
 ///
-/// <viewPrimaryConstructor> ::=
-///   '(' <type> <identifier> ')'
+/// <inlineMemberDeclaration> ::= <classMemberDefinition>
 ///
-/// <viewMemberDeclaration> ::=
-///   <classMemberDefinition>
-/// ...
-/// If a view declaration named View includes a <viewPrimaryConstructor> then it
-/// is a compile-time error if the declaration includes a constructor
-/// declaration named View. (But it can still contain other constructors.)
-///
-/// @description Checks that if a view declaration named `View` includes a
-/// <viewPrimaryConstructor> then it can contain other (than named `View`)
-/// constructors with optional parameters
+/// @description Checks that an inline class declaration may contain constant
+/// constructors
 ///
 /// @author sgrekhov22@gmail.com
 
 // SharedOptions=--enable-experiment=inline-class
 
 import "../../Utils/expect.dart";
 
-view class View(int id) {
-  View.c1(int x, {int y = 0}) : id = x + y;
-  View.c2(int x, [int y = 0]) : id = x + y;
-  View.c3([int x = 0]) : id = x;
-  View.c4({int x = 1}) : id = x;
+inline class IC {
+  final int id;
```

Comment:
Would be good to have the plain `const IC(this.id);` again here. Do we have an error test (where the constant constructor has an initializer list containing an expression which isn't potentially constant)?

Reply:
Added `syntax_A02_t06.dart` testing it